### PR TITLE
Update clinvar import

### PR DESCRIPTION
**Pull request by ima23**
- add functionality to match based on location and alleles existing rsIDs in the DB for RCVs without rsIDs
- update clean option to remove previously imported variation/variation_feature from ClinVar
- add try/catch for update sql warnings
- don't store risk_alleles with length > 255: they were being truncated
- handle clinsigs not present in the variation clinsig defined set

### CODE REVIEW

File: scripts/import/import_clinvar_xml [hunk 1]
```diff
@@ -1195,10 +1249,19 @@ sub clean_old_data{
...
+    my $var_recs_del_sth    = $dba->dbc->do(qq[ delete from $table where variation_id in
+                                                    (select variation_id from variation v, source s
+                                                     where v.source_id = s.source_id and s.name ='ClinVar')
+                                                    ]);
+  }
+  my $var_feat_del_sth    = $dba->dbc->do(qq[ delete from variation_feature where source_id in (select source_id from source where name ='ClinVar'); ]);
+  my $var_del_sth        = $dba->dbc->do(qq[ delete from variation where source_id in (select source_id from source where name ='ClinVar'); ]);
+  my $synonym_del_sth    = $dba->dbc->do(qq[ delete from variation_synonym where source_id in (select source_id from source where name ='OMIM' ) ]);
 
+  my $var_updt_sth       = $dba->dbc->do(qq[ update variation set clinical_significance=NULL ]);
```
Comment (sarahhunt) [inline scripts/import/import_clinvar_xml:1264]: This risks the clinical significance info in variation and variation_feature becoming out of sync

**Comment by ima23** [comment]
Thanks @sarahhunt! The PR is ready for review.

File: scripts/import/import_clinvar_xml [hunk 2]
```diff
@@ -1208,7 +1271,7 @@ sub usage{
     die "\n\tUsage: import_clinvar_xml -data_file [ClinVar xml] -registry [registry file] -assembly [GRCh37/GRCh38]
 
 \t\toptions: -structvar  (only import ClinVar statuses for structural variations)
-\t\toptions: -clean      ( delete old phenotype_feature, phenotype_feature_attrib, variation_set_variation (OMIM set), clinical_significance and synonym data)
+\t\toptions: -clean      ( delete old phenotype_feature, phenotype_feature_attrib, variation(ClinVar), variation_feture (ClinVar), variation_set_variation (OMIM set), clinical_significance and synonym data)
 \n\n";
```
[the remark this answers is not in the document]
Comment (sarahhunt) [reply scripts/import/import_clinvar_xml:1321]: This comment could be misinterpreted to suggest all records are removed for the ClinVar, OMIM, and All phenotypes set, when all set records are removed for variants with a source of ClinVar.

File: scripts/import/import_clinvar_xml [hunk 3]
```diff
@@ -672,33 +690,52 @@ sub import{
...
-      import_phenotype_feature($record, $feature_object, 'Variation', $feat, $alt_allele);
     }
   }
-  elsif(defined  $record->{feature_info}->{dbVar}){
+  elsif(!defined $structvar && $try_match ){
+    ($feature_object, $feat, $alt_allele) = get_variant($record, undef);
+    next unless defined $feature_object;
+
+    add_clinvar_data($feature_object, $feat, $alt_allele, $record);
+  }
```
Comment (sarahhunt) [inline scripts/import/import_clinvar_xml:705]: When I test with a variant that isn't in the database, I get these messages:

Exiting subroutine via next at scripts/import/import_clinvar_xml line 702.
Exiting eval via next at scripts/import/import_clinvar_xml line 702.
Exiting subroutine via next at scripts/import/import_clinvar_xml line 702.

They aren't very useful and could make it hard to spot other warnings. It's more useful to list the variants which could not be found.

**Pull request merged**